test/e2e/performanceprofile/functests/utils/client/clients.go

```diff
@@ -67,29 +75,62 @@ func init() {
 	Client, err = New()
 	if err != nil {
 		testlog.Info("Failed to initialize client, check the KUBECONFIG env variable", err.Error())
-		ClientsEnabled = false
+		return
+	}
+
+	ControlPlaneClient, err = NewControlPlane()
+	if err != nil {
+		testlog.Info("Failed to initialize ControlPlaneClient client, check the KUBECONFIG env variable", err.Error())
+		return
+	}
+	DataPlaneClient, err = NewDataPlane()
+	if err != nil {
+		testlog.Info("Failed to initialize DataPlaneClient client, check the KUBECONFIG env variable", err.Error())
 		return
 	}
 	K8sClient, err = NewK8s()
 	if err != nil {
 		testlog.Info("Failed to initialize k8s client, check the KUBECONFIG env variable", err.Error())
-		ClientsEnabled = false
 		return
 	}
 	ClientsEnabled = true
 }
 
-// New returns a controller-runtime client.
 func New() (client.Client, error) {
 	cfg, err := config.GetConfig()
 	if err != nil {
 		return nil, err
 	}
+	return client.New(cfg, client.Options{})
+}
 
-	c, err := client.New(cfg, client.Options{})
+// NewControlPlane returns a new controller-runtime client for ControlPlaneClient cluster.
+func NewControlPlane() (client.Client, error) {
+	c, err := New()
```

Comment:
nit: outside hypershift this is wasted, we should do like `NewDataPlane()` and create this client only if needed - or document why doing like this is fine

Reply:
Right. Thanks!